Description: Create the doc/mpl_examples symlink if not existing.
Author: Michael Droettboom
Origin: upstream, https://github.com/matplotlib/matplotlib/pull/2187
Applied-Upstream: https://github.com/mdboom/matplotlib/commit/63405ffaa3fb164950d79408855485f5c4b3d9f2,
 https://github.com/mdboom/matplotlib/commit/2c5ef7beabb4894e8c146fe002fc36ff5d47004e
Last-Update: 2013-07-17
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/doc/make.py
+++ b/doc/make.py
@@ -236,6 +236,19 @@ current_dir = os.getcwd()
 os.chdir(os.path.dirname(os.path.join(current_dir, __file__)))
 copy_if_out_of_date('../INSTALL', 'users/installing.rst')
 
+# Create the examples symlink, if it doesn't exist
+
+required_symlinks = [
+    ('mpl_examples', '../examples/'),
+    ('mpl_toolkits/axes_grid/examples', '../../../examples/axes_grid/')
+    ]
+
+for link, target in required_symlinks:
+    if not os.path.exists(link):
+        if hasattr(os, 'symlink'):
+            os.symlink(target, link)
+        else:
+            shutil.copytree(os.path.join(link, '..', target), link)
 
 if len(sys.argv)>1:
     if '--small' in sys.argv[1:]:
